services/QuillLMS/app/queries/quill_big_query/concerns/generated_view_query_fallback.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module QuillBigQuery
+  module Concerns
+    module GeneratedViewQueryFallback
+      extend ActiveSupport::Concern
+
+      class BrokenViewError < StandardError; end
+
+      BROKEN_VIEW_ERRORS = [
+        ::Google::Cloud::NotFoundError,
+        ::Google::Cloud::InvalidArgumentError
+      ]
+
+      def run_query = run_query_with_fallback
+
+      def run_query_with_fallback
+        query_runner(query)
+      rescue *BROKEN_VIEW_ERRORS => e
+        ErrorNotifier.report(BrokenViewError.new(e.message))
+        query_runner(query_fallback)
+      end
+
+      private def query_runner(query)
+        post_query_transform(runner.execute(query))
```

Comment:
`post_query_transform` has previously been guaranteed to exist via inheritance from `query.rb`. Including this concern in a non-descendent of `Query` would cause an error.

Reply:
That's a good point, let me think about this. I originally had a `def post_query_transform = (raise NotImplementedError) but it was not ideal since it forced me to move the includes further up in the hierarchy than I wanted to avoid overwriting a base class method. This will be a loud error if the class doesn't have it, so I'm not too worried, but let me think on this.